src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_OUTPUT_FORMAT.getName(), checkForNullData(decoderStats.getVideoOutputFormat()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_FRAMING.getName(), checkForNullData(decoderStats.getVideoFraming()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_SLICES_PER_FRAME.getName(), checkForNullData(decoderStats.getVideoSlicesPerFrame()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_INPUT_FRAME_RATE.getName(), checkForNullData(decoderStats.getVideoInputFrameRate()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.STILL_IMAGE.getName(), checkForNullData(decoderStats.getStillImage()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_DECODER_STATE.getName(), checkForNullData(decoderStats.getVideoDecoderState()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.VIDEO_DISPLAY_FORMAT.getName(), checkForNullData(decoderStats.getVideoDisplayFormat()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.LOAD_PERCENTAGE.getName(), checkForNullData(decoderStats.getLoadPercentage().toString()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.PREPROCESSOR_STATE.getName(), checkForNullData(decoderStats.getPreprocessorState()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.TROUBLE_CODE.getName(), checkForNullData(decoderStats.getTroubleCode().toString()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.DISPLAYED_OUTPUT_FRAMES.getName(), checkForNullData(decoderStats.getDisplayedOutputFrames()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.SKIPPED_OUTPUT_FRAMES.getName(), checkForNullData(decoderStats.getSkippedOutputFrames()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.REPLAYED_OUTPUT_FRAMES.getName(), checkForNullData(decoderStats.getReplayedOutputFrames()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_STATE.getName(), checkForNullData(decoderStats.getAudioState()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_SAMPLE_RATE.getName(), checkForNullData(decoderStats.getAudioSampleRate()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_PAIRS_AMOUNT.getName(), checkForNullData(decoderStats.getAudioPairsAmount().toString()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_DECODED_FRAMES.getName(), checkForNullData(decoderStats.getAudioDecodedFrames()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_PLAYED_FRAMES.getName(), checkForNullData(decoderStats.getAudioPlayedFrames()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.AUDIO_SKIPPED_FRAMES.getName(), checkForNullData(decoderStats.getAudioSkippedFrames()));
+
+					if (audioPairs != null) {
+						for (int i = 1; i < audioPairs.size(); i++) {
```

Comment:
please add the comment into the code to explain why we ignore the first element in audioPairs

Reply:
My Mistake, It should be started with index 0, nothing is ignored here.